plasmapy/formulary/parameters.py

```diff
@@ -663,23 +689,27 @@ def thermal_speed(
     >>> thermal_speed(1e6*u.K, "e-", method="mean_magnitude")
     <Quantity 621251... m / s>
     """
-    m = mass if np.isfinite(mass) else particles.particle_mass(particle)
+    if mass is None:
+        mass = particles.particle_mass(particle)
 
     # different methods, as per https://en.wikipedia.org/wiki/Thermal_velocity
     try:
-        coef = _coefficients[ndim]
+        coeff = thermal_speed_coefficients[ndim]
```

Comment:
Just taking the opportunity to avoid overwriting the var name later with a minor refactor
```suggestion
        coeffs_for_dim = thermal_speed_coefficients[ndim]
```